### PR TITLE
Clean up some docs for 4.x

### PR DESCRIPTION
**Pull request by markstory**
* Fix version numbers.
* Remove duplicate content for requests/responses.
* Fix middleware() hook signatures.
* Fix plugin middleware example.
* Remove outdated warning for http/client.

### CODE REVIEW

File: en/controllers/middleware.rst [hunk 1]
```diff
@@ -68,15 +68,16 @@ process, you can use the ``MiddlewareQueue`` object to attach middleware::
     namespace App;
 
     use Cake\Http\BaseApplication;
+    use Cake\Http\MiddlewareQueue;
     use Cake\Error\Middleware\ErrorHandlerMiddleware;
 
     class Application extends BaseApplication
     {
-        public function middleware($middlewareQueue)
+        public function middleware(MiddlewareQueue $middleware): MiddlewareQueue
```
Comment (dereuromark) [inline en/controllers/middleware.rst]: Cant we keep the Queue prefix for clarity?
$middleware->add(Middleware) sounds wrong to me.

File: en/tutorials-and-examples/cms/tags-and-users.rst [hunk 2]
```diff
@@ -376,9 +376,9 @@ replace the existing ``tags._ids`` control with the following::
     echo $this->Form->control('tag_string', ['type' => 'text']);
     
 We'll also need to update the article view template. In
-**src/Template/Articles/view.ctp** add the line as shown::
+**src/Template/Articles/view.php** add the line as shown::
 
-    <!-- File: src/Template/Articles/view.ctp -->
+    <!-- File: src/Template/Articles/view.php -->
```
Comment (dereuromark) [inline en/tutorials-and-examples/cms/tags-and-users.rst]: templates/Articles now right?

File: en/installation.rst [hunk 3]
```diff
@@ -248,6 +248,8 @@ production setup will look like this on the file system::
         logs/
         plugins/
         src/
+        resources/
+        templates/
```
Comment (garas) [inline en/installation.rst]: /resources
      /src 
      /templates

File: en/tutorials-and-examples/cms/tags-and-users.rst (hunk 2 above)
Comment (markstory) [reply en/tutorials-and-examples/cms/tags-and-users.rst]: Yep, missed that as I was focused on ctp.

File: en/controllers/middleware.rst (hunk 1 above)
Comment (markstory) [reply en/controllers/middleware.rst]: Sure. I was trying to pare down the line lengths.

**Pull request merged**